### PR TITLE
#3091 Exciter_INIT more widespread, pu base, alphabetical order

### PR DESCRIPTION
**Pull request by ErwanGuichard49**

### CODE REVIEW

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/Scrx.mo [hunk 1]
```diff
@@ -30,15 +30,17 @@ model Scrx "Bus-fed or solid-fed exciter model, with no excitation resistance"
...
   Modelica.Blocks.Interfaces.RealInput UOelPu(start = 0) "Overexcitation limitation output voltage in pu (base UNom)" annotation(
-    Placement(visible = true, transformation(origin = {-180, 40}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-120, 80}, extent = {{-20, -20}, {20, 20}}, rotation = 0)));
+    Placement(visible = true, transformation(origin = {-180, 80}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-120, 80}, extent = {{-20, -20}, {20, 20}}, rotation = 0)));
   Modelica.Blocks.Interfaces.RealInput UPssPu(start = 0) "Power system stabilizer output voltage in pu (base UNom)" annotation(
-    Placement(visible = true, transformation(origin = {-180, -40}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-120, -80}, extent = {{-20, -20}, {20, 20}}, rotation = 0)));
+    Placement(visible = true, transformation(origin = {-180, 0}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-120, -80}, extent = {{-20, -20}, {20, 20}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealInput UPu(start = U0Pu) "Stator voltage in pu (base UNom)" annotation(
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/Scrx.mo:36]: Do you mean terminal?

I am confused: the stator voltage is the terminal voltage. Which terminal you are talking about? This is in the documentation?

File: dynawo/sources/Models/Modelica/Dynawo/Examples/ENTSOE/TestCase1.mo [hunk 2]
```diff
@@ -118,7 +126,7 @@ equation
     Line(points = {{38, 10}, {70, 10}, {70, 6}, {78, 6}}, color = {0, 0, 127}));
   connect(generatorSynchronous.omegaPu_out, pss.omegaPu) annotation(
     Line(points = {{38, -6}, {78, -6}}, color = {0, 0, 127}));
-  connect(pss.UpssPu, avr.UpssPu) annotation(
+  connect(pss.VPssPu, avr.UpssPu) annotation(
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/ENTSOE/TestCase1.mo:129]: I didn't find VPssPu variable.

We can keep it UpssPu

File: dynawo/sources/Models/Modelica/Dynawo/Examples/ENTSOE/TestCase2.mo [hunk 3]
```diff
@@ -125,7 +133,7 @@ equation
     Line(points = {{38, 18}, {118, 18}}, color = {0, 0, 127}));
   connect(governor.PmPu, generatorSynchronous.PmPu_in) annotation(
     Line(points = {{102, -30}, {110, -30}, {110, -51}, {32, -51}, {32, -16}}, color = {0, 0, 127}));
-  connect(const.y, avr.UpssPu) annotation(
+  connect(pss.VPssPu, avr.UpssPu) annotation(
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/ENTSOE/TestCase2.mo:136]: There is no PSS in this test case

File: dynawo/sources/Models/Modelica/Dynawo/Examples/ENTSOE/TestCase3.mo [hunk 4]
```diff
@@ -148,7 +156,7 @@ equation
     Line(points = {{38, 10}, {70, 10}, {70, 6}, {78, 6}}, color = {0, 0, 127}));
   connect(generatorSynchronous.omegaPu_out, pss.omegaPu) annotation(
     Line(points = {{38, -6}, {78, -6}}, color = {0, 0, 127}));
-  connect(pss.UpssPu, avr.UpssPu) annotation(
+  connect(pss.VPssPu, avr.UpssPu) annotation(
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/ENTSOE/TestCase3.mo:159]: same for this variable

**Review by joyelfeghali — CHANGES_REQUESTED** [review]

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/Scrx.mo (hunk 1 above)
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/Scrx.mo:36]: I have replaced the description by : "Voltage amplitude at terminal in pu (base UNom)"

In GeneratorSynchronous.mo, I read 

UPu = ComplexMath.'abs'(terminal.V);
uStatorPu = 1 / rTfoPu * (terminal.V - terminal.i * Complex(RTfoPu, XTfoPu) * SystemBase.SnRef / SNom);
UStatorPu.value = ComplexMath.'abs'(uStatorPu);

Therefore UPu = UsPu <=> RTfoPu = XTfoPu = 0 and rTfoPu = 1

File: dynawo/sources/Models/Modelica/Dynawo/Examples/ENTSOE/TestCase1.mo (hunk 2 above)
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/Examples/ENTSOE/TestCase1.mo:129]: Indeed, this belongs to the branch 3091_use_new_standard_regulations

File: dynawo/sources/Models/Modelica/Dynawo/Examples/ENTSOE/TestCase2.mo (hunk 3 above)
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/Examples/ENTSOE/TestCase2.mo:136]: Indeed

File: dynawo/sources/Models/Modelica/Dynawo/Examples/ENTSOE/TestCase3.mo (hunk 4 above)
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/Examples/ENTSOE/TestCase3.mo:159]: Change reversed

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/Scrx.mo (hunk 1 above)
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/Scrx.mo:36]: In 1024625_Common Information Model _CIM_ for Dynamics_ Standard Models.pdf, the following diagram can be found :
image

Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/Scrx.mo:36]: Vc is the filtered stator voltage, Vs the PSS output voltage, Vt the voltage at the terminal.

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/IEEET1.mo [hunk 5]
```diff
@@ -67,10 +67,10 @@ model IEEET1 "IEEE type 1 Exciter (IEEET1)"
     Placement(visible = true, transformation(origin = {90, -80}, extent = {{10, -10}, {-10, 10}}, rotation = 0)));
 
   parameter Types.VoltageModulePu Efd0Pu "Initial excitation voltage in pu (user-selected base voltage)";
-  parameter Types.VoltageModulePu UStator0Pu "Initial stator voltage in pu (base UNom)";
+  parameter Types.VoltageModulePu Us0Pu "Initial stator voltage in pu (base UNom)";
```
[the remark this answers is not in the document]
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/IEEET1.mo:70]: I have completed update1.7.0.0.py.

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/Scrx.mo (hunk 1 above)
Comment (joyelfeghali) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/Scrx.mo]: In our synchronous generator model, we added this transformer, but usually, we never use it, so UPu = UsPu all the time.

When we have standard regulations, they always define a generator model without a transformer. The terminal of the generator model is always the stator voltage. 
Unless they define it with words that Vt is the voltage behind the transformer, Vt is the stator voltage.

We can keep it as it is

Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/Scrx.mo]: Thanks for the explanation. I am reversing this change.

File: util/updateXML/update1.7.0.0.py [hunk 6]
```diff
@@ -42,3 +42,17 @@ def update(jobs):
...
         for unit_dynamic_model in unit_dynamic_models:
             unit_dynamic_model.set_name("Dynawo.Electrical.Controls.Machines.VoltageRegulators.Standard.SCRX1")
+
+    modelica_models = jobs.dyds.get_modelica_models(lambda modelica_model: modelica_model.get_id() == "Dynawo.Electrical.Controls.Machines.VoltageRegulators.Standard.IEEET1")
+    for modelica_model in modelica_models:
+        unit_dynamic_models1 = modelica_model.get_unit_dynamic_models(lambda unit_dynamic_model: unit_dynamic_model.get_id() == "voltageRegulator")
+        for unit_dynamic_model1 in unit_dynamic_models1:
+            unit_dynamic_model1.init_connects.change_var_name("UStator0Pu", "Us0Pu")
+
+    modelica_models = jobs.dyds.get_modelica_models(lambda _: True)
```
Comment (joyelfeghali) [inline util/updateXML/update1.7.0.0.py:52]: Could you add also the one of model_template? Thanks

**Review by joyelfeghali — CHANGES_REQUESTED** [review]

File: util/updateXML/update1.7.0.0.py (hunk 6 above)
Comment (ErwanGuichard49) [reply util/updateXML/update1.7.0.0.py:52]: I have just pushed a new version with an additional model_template.

File: util/updateXML/update1.7.0.0.py [hunk 7]
```diff
@@ -42,3 +42,24 @@ def update(jobs):
         )
         for unit_dynamic_model in unit_dynamic_models:
             unit_dynamic_model.set_name("Dynawo.Electrical.Controls.Machines.VoltageRegulators.Standard.SCRX1")
+
+    modelica_models = jobs.dyds.get_modelica_models(lambda modelica_model: modelica_model.get_id() == "Dynawo.Electrical.Controls.Machines.VoltageRegulators.Standard.IEEET1")
+    for modelica_model in modelica_models:
+        unit_dynamic_models1 = modelica_model.get_unit_dynamic_models(lambda unit_dynamic_model: unit_dynamic_model.get_id() == "voltageRegulator")
+        for unit_dynamic_model1 in unit_dynamic_models1:
+            unit_dynamic_model1.init_connects.change_var_name("UStator0Pu", "Us0Pu")
```
Comment (joyelfeghali) [inline util/updateXML/update1.7.0.0.py:50]: It is a parameter , I think we  can use something similar to that : (like what we did with HVDC)

IEEET1 = job.dyds.get_bbms(lambda bbm: bbm.get_lib_name() == "IEEET1")
for voltageregulator in IEEET1:
        value =voltageregulator.parset.get_param_value("voltageregulator_UStator0Pu")
        voltageregulator.parset.remove_param_or_ref("voltageregulator_UStator0Pu")
        voltageregulator.parset.add_param("DOUBLE", "voltageregulator_Us0Pu", value)

to be tested

Comment (ErwanGuichard49) [reply util/updateXML/update1.7.0.0.py:50]: UStator0Pu appears in initConnect statements, the structure used for HVDC parameters is aimed at renaming parameters in .par.

Comment (joyelfeghali) [reply util/updateXML/update1.7.0.0.py:50]: Okay could you add also one of of model_template?

Comment (ErwanGuichard49) [reply util/updateXML/update1.7.0.0.py:50]: Done

Comment (ErwanGuichard49) [reply util/updateXML/update1.7.0.0.py:50]: The test nrt-update 1.6.0 to 1.7.0 is successful.

**Review by joyelfeghali — APPROVED** [review]

**Pull request merged**